lib/plug/builder.ex

```diff
@@ -241,4 +247,21 @@ defmodule Plug.Builder do
       end
     end
   end
+
+  defp log_halt(plug_type, plug, builder_opts) do
+    if level = builder_opts[:log_on_halt] do
+      quote do
+        plug = unquote(plug)
+        message = case unquote(plug_type) do
```

Comment:
I agree, but if I do that I don't know how to access the `__ENV__`. Little help here? :)